### DISCUSSION TITLE
What I should learn to contribute?

### DISCUSSION THREAD

**Discussion by freedxmgxd** [carbon-language/carbon-lang#1749] [Getting started]
What should I learn first in order to be able to contribute to the development of carbon? Assuming that's I completely new in the field.

**Comment by OlaFosheimGrostad** [comment]
It says in [CONTRIBUTING.md](https://github.com/carbon-language/carbon-lang/blob/trunk/CONTRIBUTING.md#ways-to-contribute): «If you're looking for a quick way to contribute, commenting on proposals is a way to provide proposal authors with a breadth of feedback. The "Evolution > Ideas" forum is where authors will go for early, high-level feedback. The "Evolution > Proposal reviews" forum will have more mature proposals that are nearing the decision process.»

But where are these forums? I don't know…

**Reply by freedxmgxd** [reply]
I want to contribute to code too, but thanks for the link to CONTRIBUTING.md, I have not seen them yet.

**Reply by OlaFosheimGrostad** [reply]
Ah, yes, well. I wish I there was a way to build the interpreter without bazelisk and homebrew. Then I might give it a shot…

Maybe they need more test examples? I guess that could be a starting point? But then again, maybe they need them to be written for fuzzing… which apparently requires the bazel monster…

Hm…

**Reply by josh11b** [reply]
Sorry, looks like the contributing doc is out of date with respect to the forums. I asked @jonmeow to update them: https://discord.com/channels/655572317891461132/994710526674346085/1001564468922298368

**Comment by YagaoDirac** [comment]
If you are completely new to programming, you still have a way to contribute. Since a lot proposal are made by people who are not very experience, and sometimes this can be confusing, if you are completely new, when you discuss, remind people of yourself being super new to program, and then it's way easier for others to know how to treat ideas better. 
This project is not aimed to be new programmer friendly. But when I read the discussions I was surprised by the polls. The real problem this project is to solve is almost ignored. People care only about the very trivial grammar part. 
So, if you are completely new and willing to contribute, try learn as much and as fast, train more, and read the discussions and tell people your idea and your preference.
Or more concretely, I think you really need to learn python, js, c++, data structure, algorithm, c++ std library, before you can do anything for this project.

**Comment by josh11b** [comment]
We are making a list of good first issues to tackle for people new to the project here:
https://github.com/carbon-language/carbon-lang/labels/good%20first%20issue
